rust/core/src/driver_manager.rs

```diff
@@ -316,6 +493,98 @@ impl Driver for ManagedDriver {
     }
 }
 
+#[cfg(windows)]
+fn load_driver_from_registry(
+    root: &windows_registry::Key,
+    driver_name: &OsStr,
+    entrypoint: Option<&[u8]>,
+) -> Result<DriverInfo> {
+    const ADBC_DRIVER_REGISTRY: &str = "SOFTWARE\\ADBC\\Drivers";
+    let drivers_key = root.open(ADBC_DRIVER_REGISTRY)?;
+
+    Ok(DriverInfo {
+        driver_name: drivers_key.get_string("name").unwrap_or_default(),
+        entrypoint: entrypoint
+            .or_else(|| drivers_key.get_string("entrypoint").map(|e| e.into_bytes())),
+        version: drivers_key.get_string("version").unwrap_or_default(),
+        source: drivers_key.get_string("source").unwrap_or_default(),
+        lib_path: PathBuf::from(drivers_key.get_string("driver").unwrap_or_default()),
+    })
+}
+
+fn get_optional_key(manifest: &Table, key: &str) -> String {
+    manifest
+        .get(key)
+        .and_then(Value::as_str)
+        .unwrap_or_default()
+        .to_string()
+}
+
+fn load_driver_manifest(info: &DriverInfo) -> Result<DriverInfo> {
+    let contents = fs::read_to_string(info.manifest_file.as_path()).map_err(|e| {
+        Error::with_message_and_status(
+            format!(
+                "Could not read manifest '{}': {e}",
+                info.manifest_file.display()
+            ),
+            Status::InvalidArguments,
+        )
+    })?;
+
+    let manifest = contents
+        .parse::<Table>()
+        .map_err(|e| Error::with_message_and_status(e.to_string(), Status::InvalidArguments))?;
+
+    let driver_name = get_optional_key(&manifest, "name");
+    let version = get_optional_key(&manifest, "version");
+    let source = get_optional_key(&manifest, "source");
+    let (os, arch, extra) = arch_triplet();
+
+    let mut lib_path = PathBuf::default();
+    if let Some(driver) = manifest.get("Driver").and_then(|v| v.get("shared")) {
+        if driver.is_str() {
+            lib_path = PathBuf::from(driver.as_str().unwrap_or_default());
+        } else if driver.is_table() {
+            lib_path = PathBuf::from(
+                driver
+                    .get(format!("{os}_{arch}{extra}"))
+                    .and_then(Value::as_str)
+                    .unwrap_or_default(),
+            );
+        }
+    }
+
+    if lib_path.as_os_str().is_empty() {
+        return Err(Error::with_message_and_status(
+            format!(
+                "Manifest '{}' missing appropriate path for current arch",
```

Comment:
Maybe we can include a hint (e.g. the path) to fix it?

Reply:
updated the error message